test/modules/core/shadertools/fp64/fp64-arithmetic-transform.spec.js

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2015 - 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the 'Software'), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import test from 'tape-catch';
+import {runTests} from './fp64-test-utils-transform';
+import {glGetDebugInfo} from 'luma.gl';
+import {fixture} from 'luma.gl/test/setup';
+const gl = fixture.gl2;
+
+ // Failing tests cases are ignred based ib cyrrebt gpu and glslFunc, using ignoreFor field
+ // ignoreFor: [{gpu: ['glslFun']}] => ignores glslFun when running `gpu`
+const commonTestCases = [
+  {a: 3.0e-19, b: 3.3e+13},
+  {a: 9.9e-40, b: 1.7e+3},
+  {a: 1.5e-36, b: 1.7e-16},
+  {a: 9.4e-26, b: 51},
+  {a: 6.7e-20, b: 0.93},
+
+  // mul_fp64: Large numbers once multipled, can't be represented by 32 bit precision and Math.fround() returns NAN
+  // sqrt_fp64: Fail on INTEL with margin 3.906051071870294e-12
+  {a: 2.4e+3, b: 5.9e+31, ignoreFor: {all: ['mul_fp64'], Intel: ['sqrt_fp64']}},
```

Comment:
Nit: maybe lower case gpu vendor names `intel`? Maybe easier to not have to remember capitalizations?